### PR TITLE
[PRO-1648] Add more id-nr in system_info

### PR DESCRIPTION
**Pull request by Danten**
I also changed so we add the id-nr when fetching rather than when
inserting into the database. The idea with doing it before inserting
was that it would not perform this computation all the time. But this
interferes with group_info, since that shouldn't consider id-nr.

### CODE REVIEW

**Comment by simao** [comment]
Why does this interfere with group info though? Maybe group info should ignore that field. We should not be computing this all the time.

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala [hunk 1]
```diff
@@ -54,13 +54,15 @@ class GroupsResource(namespaceExtractor: Directive1[Namespace], deviceNamespaceA
...
+      allDevices <- list(ns)
+      matchingDevices = allDevices.collect {
+        case info if compare(disregard(removeIdNrs(info.systemInfo), discarded), matching)
+                       .equals((matching, Json.Null))
+          => info.uuid
+      }
+      res <- DBIO.sequence(matchingDevices.map { device =>
+                             GroupMemberRepository.addOrUpdateGroupMember(groupId, device)
+                           })
     } yield res
```
Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: Think could be more readable if `yield matchingDevices.map { device => GroupMemberRepository.addOrUpdateGroupMember(groupId, device) }` and then do the sequence outside the for

Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: Also this whole code should not be in this resource this is getting to complicated. This should all be extracted to a separate class that handles all this mess and can be tested separatly if needed

Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: In general if it does not need to handle anything from `akka.http.scaladsl.*` then it should not be in the controller

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala [hunk 2]
```diff
@@ -72,8 +74,8 @@ class GroupsResource(namespaceExtractor: Directive1[Namespace], deviceNamespaceA
 
   def createGroupFromDevices(request: CreateGroupRequest, namespace: Namespace): Route = {
     val commonJsonDbIo = for {
-      info1 <- findByUuid(request.device1)
-      info2 <- findByUuid(request.device2)
+      info1 <- findByUuid(request.device1).map(removeIdNrs)
+      info2 <- findByUuid(request.device2).map(removeIdNrs)
```
Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: This method should also not be on the controller, should be somewhere else that knows about `removeIdNrs` and databases

**Comment by jerrytrieu** [comment]
Should be PRO-1684

**Pull request closed without merging**